src/pilot/assign_role.py

```diff
@@ -1014,34 +1025,67 @@ def generate_osd_config(ip_mac_service_tag, drac_client):
 def get_drives(drac_client):
     spinners = []
     ssds = []
-    physical_disks = drac_client.list_physical_disks()
-    for physical_disk in physical_disks:
-        # Eliminate physical disks that in a state other than non-RAID
-        if physical_disk.raid_status != "non-RAID":
-            LOG.info("Skipping disk {id}, because it has a RAID status of "
-                     "{raid_status}".format(
-                         id=physical_disk.id,
-                         raid_status=physical_disk.raid_status))
-            continue
-
-        # Eliminate physical disks that have an error status
-        if physical_disk.status == 'error':
-            LOG.warning("Not using disk {id}, because it has a status of "
-                        "{status}".format(id=physical_disk.id,
-                                          status=physical_disk.status))
-            continue
-
-        # Go ahead and use any physical drive that's not in an error state, but
-        # issue a warning if it's not in the ok or unknown state
-        if physical_disk.status != 'ok' and physical_disk.status != 'unknown':
-            LOG.warning("Using disk {id}, but it has a status of \""
-                        "{status}\"".format(id=physical_disk.id,
-                                            status=physical_disk.status))
-
-        if physical_disk.media_type == "hdd":
-            spinners.append(physical_disk)
+    virtual_disks = drac_client.list_virtual_disks()
+
+    raid0_disks = [vd for vd in virtual_disks if vd.raid_level != '1']
+    raid1_disks = []
+    for vd in virtual_disks:
+        if vd.raid_level == '1':
+            raid1_disks.extend(vd.physical_disks)
+
+    # Getting all physical disks of raid_disks except RAID1 disks
+    physical_disks = {pd.id: pd for pd in drac_client.list_physical_disks()
+                      if pd.id not in raid1_disks}
+
+    for virtual_disk in raid0_disks:
+        phy_disks = [physical_disks[pd_id] for pd_id
+                     in virtual_disk.physical_disks]
+        if phy_disks[0].media_type == 'hdd':
+            spinners.append(virtual_disk)
         else:
-            ssds.append(physical_disk)
+            ssds.append(virtual_disk)
+
+        {physical_disks.pop(pd.id) for pd in phy_disks}
+
+    if physical_disks:
+        for pd_id in physical_disks:
+            # Eliminate physical disks that in a state other than non-RAID
+            if physical_disks[pd_id].raid_status != "non-RAID":
+                LOG.info("Skipping disk {id}, because it has a RAID status of "
+                         "{raid_status}".format(
+                             id=physical_disks[pd_id].id,
+                             raid_status=physical_disks[pd_id].raid_status))
+                continue
+
+            # Eliminate physical disks that have a failed status
+            if physical_disks[pd_id].raid_status == "failed":
```

Comment:
This entire "if" clause will never be entered because the "if" clause on line 1053 above will be entered if raid_status=="failed".  Let's just remove all the code from 1060 through 1066 since it won't be used, and modify the comment on line 1052 above to: "Eliminate physical disks in a state other than non-RAID including failed disks"